code/modules/RCD/schematics/clockwork.dm

```diff
@@ -0,0 +1,77 @@
+#define CLOCK_AIRLOCK_CONVERT_COST 4
+#define CLOCK_FLOOR_CONVERT_COST   1
+#define CLOCK_WALL_CONVERT_COST    3
+
+/datum/rcd_schematic/clock_door
+	name = "Door"
+	category = "Clockwork"
+
+/datum/rcd_schematic/clock_door/attack(var/atom/A, var/mob/user)
+	if(!istype(A, /turf))
+		return TRUE
+	if(locate(/obj/machinery/door/airlock/clockwork) in A)
+		return "there is already a door on this spot!"
+	to_chat(user, "Building door...")
+	if(!do_after(user, A, 50))
```

Comment:
You need to repeat the first sanity check from above after this. I also suggest changing this `do_after` to `master.delay`, and perhaps 50 to `5 SECONDS`.